modyn/supervisor/internal/pipeline_executor/pipeline_executor.py

```diff
@@ -305,17 +317,16 @@ def _handle_triggers_within_batch(self, batch: list[tuple[int, int, int]], trigg
                 )
             else:
                 logger.info(f"Skipping training on empty trigger {trigger_id}]")
-            self._persist_pipeline_log()
 
             self.num_triggers = self.num_triggers + 1
             if self.maximum_triggers is not None and self.num_triggers >= self.maximum_triggers:
-                break
+                return num_triggers
```

Comment:
just nitpicking: I think `num_triggers` is not needed as we can replace every occurrence of it with `len(triggering_idx_list)` right?